applications/nrf_desktop/doc/leds.rst

```diff
@@ -0,0 +1,43 @@
+.. leds:
+
+leds module
+###########
+
+``leds`` module is responsible for controlling LED diodes (setting LED effect in reaction to ``led_event``).
+
+
+Configuration
+*************
+
+For setting brightness ``leds`` uses PWM driver.
+In order to use the module, following configuration options have to be set:
+
+- :option:`CONFIG_PWM`
+- :option:`CONFIG_PWM_0`
+  0 denotes PWM port number. All PWM ports used by application should be configured.
+
+For current implementation :option:`CONFIG_PWM_NRFX` is set and :option:`CONFIG_PWM_NRF5_SW` is unset (in order to use hardware PWM driver).
+
+Module is enabled and configured using Kconfig options (details described as help).
+
+File ``leds_def.h`` is used to define mapping PWM channels to pin numbers.
+Current implementation uses symbols defined in device tree.
+These symbols are defined for board, but application may overlay definining own ``dts.overlay`` file in the application configuration directory dedicated for this board.
+
+
+Implementation details
+**********************
+
+Module sets effect for given LED based on ``led_event``.
+In addition, it turns off all LEDs when going to power down state.
+In such case, PWM drivers are also set to suspended state (in order to reduce power consumption).
+While application goes to error state, LEDs are still used to indicate error.
+
+To achieve selected LED effect, colors for given LED are periodically updated using work (:c:type:`k_delayed_work`).
+One work updates color of single LED.
+
+:c:type:`led_effect` contains pointer to array of :c:type:`led_effect_step`.
+During every update, next LED color is calculated using linear approximation between current LED color and color described in next step.
```

Comment:
```suggestion
During every update, the next LED color is calculated using a linear approximation between the current LED color and the color described in next step.
```
`described` by what?

Reply:
Actually color is described by 3 numbers representing R,G,B. For monochromatic LED you get only one number representing brightness.